templates/repo/issue/branch_selector_field.tmpl

```diff
@@ -1,11 +1,11 @@
 {{if and (not .Issue.IsPull) (not .PageIsComparePull)}}
 <input id="ref_selector" name="ref" type="hidden" value="{{.Issue.Ref}}">
-<input id="editing_mode" name="edit_mode" type="hidden" value="{{(or .IsIssueWriter .HasIssuesOrPullsWritePermission)}}">
+<input id="editing_mode" name="edit_mode" type="hidden" value="{{.HasIssuesOrPullsWritePermission}}">
 <form method="POST" action="{{$.RepoLink}}/issues/{{.Issue.Index}}/ref" id="update_issueref_form">
 	{{$.CsrfTokenHtml}}
 </form>
 
-<div class="ui {{if .ReadOnly}}disabled{{end}} floating filter select-branch dropdown" data-no-results="{{.i18n.Tr "repo.pulls.no_results"}}">
+<div class="ui {{if and .Issue (not $.IsIssuePoster) (not .HasIssuesOrPullsWritePermission)}}disabled{{end}} floating filter select-branch dropdown" data-no-results="{{.i18n.Tr "repo.pulls.no_results"}}">
```

Comment:
Are you sure about this? This implies that every PR can be edited by anyone?

Reply:
Se https://github.com/go-gitea/gitea/blob/ac88b0ee839bfbfae5759e211d0f9a69abe4d8f7/routers/repo/issue.go#L1621-L1624
* all user can choose it when create new issue .(``.Issue`` use to check whether in new issue page or not)
* all user can choose it for their own issue (``.IsIssuePoster``)
* user with issue write permission can do it for any issue. (``.HasIssuesOrPullsWritePermission``)

Reply:
Ok this seems fragile & like something we'll get wrong in future.

I wonder is there a clearer way this could be written?